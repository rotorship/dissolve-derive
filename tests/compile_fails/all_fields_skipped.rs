use dissolve_derive::Dissolve;

// Test 1: All fields skipped should fail
#[derive(Dissolve)]
struct AllSkipped {
	#[dissolved(skip)]
	field1: String,

	#[dissolved(skip)]
	field2: i32,
}

fn main() {}
