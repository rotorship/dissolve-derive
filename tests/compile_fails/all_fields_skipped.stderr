error: cannot create dissolved struct with no fields (all fields are skipped)
 --> tests/compile_fails/all_fields_skipped.rs:5:8
  |
5 | struct AllSkipped {
  |        ^^^^^^^^^^
